libslide/src/bignum/sub.rs

```diff
@@ -0,0 +1,148 @@
+#![allow(clippy::suspicious_arithmetic_impl)]
+use crate::bignum::utils::is_smaller;
+use crate::bignum::Bignum;
+use std::convert::TryInto;
+use std::mem;
+use std::ops;
+
+impl ops::Sub for Bignum {
+    type Output = Bignum;
+
+    fn sub(self, other: Bignum) -> Bignum {
+        // make the decimal vectors the same size
+        // fill lhs and rhs vectors as well
+        let mut lhs_dec: Vec<u8>;
+        let mut res_dec: Vec<u8>;
+        let self_size: usize = self.dec.len();
+        let other_size: usize = other.dec.len();
+        let (mut res_int, mut lhs_int) = if self_size > other_size {
+            res_dec = self.dec;
+            lhs_dec = vec![0; self_size];
+            lhs_dec[..other_size].clone_from_slice(&other.dec[..other_size]);
+            (self.int, other.int)
+        } else {
+            res_dec = other.dec;
+            lhs_dec = vec![0; other_size];
+            lhs_dec[..self_size].clone_from_slice(&self.dec[..self_size]);
+            (other.int, self.int)
+        };
+
+        // now we need to make the bigger number the res dec
+        if is_smaller(&lhs_dec, &res_dec, &lhs_int, &res_int) {
+            mem::swap(&mut lhs_dec, &mut res_dec);
+            mem::swap(&mut lhs_int, &mut res_int);
+        }
```

Comment:
Why not implement [Ord](https://doc.rust-lang.org/std/cmp/trait.Ord.html) for bignum instead?

Reply:
I will do this instead :)